### PR TITLE
Creates draft of an RFC proposing the GitHub access control policy

### PR DESCRIPTION
**Pull request by nellshamrell**
### [Rendered](https://github.com/rust-lang/wg-governance/blob/add-gh-access-control-rf-draft/draft-rfcs/RFC-github-access-policy.md)

Signed-off-by: Nell Shamrell <nellshamrell@gmail.com>

### CODE REVIEW

File: draft-rfcs/RFC-github-access-policy.md
Comment (BatmanAoD) [inline draft-rfcs/RFC-github-access-policy.md]: Typo:

```suggestion
Access control for the [Rust Lang GitHub Organization](https://www.github.com/rust-lang) and repositories within that organization is currently managed ad-hoc. We need a policy that defines how these accesses are granted and managed. This will allow us to have greater security in permissions to our GitHub org and also allow the infra team to build appropriate tooling to automate access control when possible.
```

File: draft-rfcs/RFC-github-access-policy.md
Comment (BatmanAoD) [inline draft-rfcs/RFC-github-access-policy.md]: I am guessing this should say "both the RustLang and <something> repositories"?

Separately, in the next sentence, "their current workflow" is probably sufficient without "that they are used to."

File: draft-rfcs/RFC-github-access-policy.md
Comment (nellshamrell) [reply draft-rfcs/RFC-github-access-policy.md]: Ty!

File: draft-rfcs/RFC-github-access-policy.md
Comment (BatmanAoD) [inline draft-rfcs/RFC-github-access-policy.md]: Is this section necessary for this particular RFC? I suppose it could refer to other open source projects' access policies, but it seems like the benefit might not be worth the level of effort.

File: draft-rfcs/RFC-github-access-policy.md
Comment (BatmanAoD) [inline draft-rfcs/RFC-github-access-policy.md:55]:
```suggestion
- Should these rules applied to RustLang affiliated repositories and organizations that are outside of the [RustLang GitHub Org](https://www.github.com/rust-lang)?
```

File: draft-rfcs/RFC-github-access-policy.md
Comment (BatmanAoD) [inline draft-rfcs/RFC-github-access-policy.md]: Is this supposed to link to https://github.com/rust-lang/team?

**Comment by BatmanAoD** [comment]
I haven't been following very closely, but generally this looks good to me.

File: draft-rfcs/RFC-github-access-policy.md
Comment (XAMPPRocky) [inline draft-rfcs/RFC-github-access-policy.md]: What automation do you see as necessary? Are there solutions to those problems that don't require automation?

File: draft-rfcs/RFC-github-access-policy.md
Comment (nellshamrell) [reply draft-rfcs/RFC-github-access-policy.md]: Aha! Good catch!

**Review by nellshamrell — COMMENTED** [review]
Thank you for the feedback @XAMPPRocky and @BatmanAoD! I have incorporated your suggestions.

File: draft-rfcs/RFC-github-access-policy.md
Comment (nikomatsakis) [inline draft-rfcs/RFC-github-access-policy.md:38]: I think we should state explicitly that we *only* want to be giving permissions to rust teams, ideally, and never individuals

**Review by nikomatsakis — CHANGES_REQUESTED** [review]

File: draft-rfcs/RFC-github-access-policy.md
Comment (nikomatsakis) [inline draft-rfcs/RFC-github-access-policy.md:6]: maybe we should also add a note (somewhere) that this  RFC was written as part of the governance WG, in collaboration with infra folks, and give links to the issue(s) where things were discussed

**Review by nikomatsakis — CHANGES_REQUESTED** [review]

**Comment by nellshamrell** [comment]
@nikomatsakis - thank you for the feedback, I have incorporated your suggestions :)

**Review by nikomatsakis — APPROVED** [review]

**Comment by XAMPPRocky** [comment]
@nellshamrell Okay, I'm going to merge this into the repo, and then if we have to make changes later we can discuss it in a new PR.

File: draft-rfcs/RFC-github-access-policy.md
Comment (emilyalbini) [inline draft-rfcs/RFC-github-access-policy.md:24]: I think we can clarify this a little more. Something like:

* All members of the core team have the right to be organization owners.
* Selected members of the infrastructure team can be organization owners, if their work requires that.

File: draft-rfcs/RFC-github-access-policy.md
Comment (emilyalbini) [inline draft-rfcs/RFC-github-access-policy.md:45]: We should remove this, or replace it with something like:

> Bot accounts controlled by the Infrastructure Team can be granted any level of access required for them to work, at the discretion of the Infrastructure Team.

**Review by emilyalbini — COMMENTED** [review]
Mostly looks good, left a few comments. We should also try to find a consistent spelling of the organization name across the RFC :)

**Comment by nellshamrell** [comment]
Thanks for the feedback @pietroalbini! I've incorporated your suggestions in #43

**Pull request merged**
